shared/src/main/java/cn/edu/tsinghua/iginx/utils/TimeUtils.java

```diff
@@ -402,19 +404,19 @@ public static long getTimeInMs(long timestamp, String timePrecision) {
         return timeInMs;
     }
 
-    public static long getTimeInNs(long timestamp, String timePrecision) {
+    public static long getTimeInNs(long timestamp, TimePrecision timePrecision) {
         long timeInNs;
         switch (timePrecision) {
-            case "s":
+            case S:
                 timeInNs = timestamp * 1000000000;
                 break;
-            case "ms":
+            case MS:
                 timeInNs = timestamp * 1000000;
                 break;
-            case "us":
+            case US:
                 timeInNs = timestamp * 1000;
                 break;
-            case "ns":
+            case NS:
                 timeInNs = timestamp;
                 break;
             default:
```

Comment:
You have defined TimePrecision to be an Enum with precisions like day, week, year. But, here, you do not have processing logic for those precisions and by default you set it to nanosecond. This is an obvious bug.